core/templates/pages/contributor-dashboard-page/contributor-stats/contributor-stats.component.ts

```diff
@@ -0,0 +1,388 @@
+    this.selectedContributionType = currentOption.displayName;
+    this.dropdownShown = false;
+    this.mobileDropdownShown = false;
+  }
+
+  async fetchStats(): Promise<void> {
+    const response = await this.contributionAndReviewStatsService.fetchAllStats(
+      this.username);
+
+    if (response.translation_contribution_stats.length > 0) {
+      response.translation_contribution_stats.map((stat) => {
+        if (!this.statsData.translationContribution[stat.language_code]) {
+          this.statsData.translationContribution[stat.language_code] = this
+            .createTranslationContributionStatsPage(stat);
+        } else {
+          this.statsData.translationContribution[stat.language_code].data.push(
+            this.createTranslationContributionStat(stat));
+        }
+      });
+    }
+
+    if (response.translation_review_stats.length > 0) {
+      response.translation_review_stats.map((stat) => {
+        if (!this.statsData.translationReview[stat.language_code]) {
+          this.statsData.translationReview[stat.language_code] = this
+            .createTranslationReviewStatsPage(stat);
+        } else {
+          this.statsData.translationReview[stat.language_code].data.push(
+            this.createTranslationReviewStat(stat));
+        }
+      });
+    }
+
+    if (response.question_contribution_stats.length > 0) {
+      this.statsData.questionContribution = this
+        .createQuestionContributionStatsPage(
+          response.question_contribution_stats);
+    }
+
+    if (response.question_review_stats.length > 0) {
+      this.statsData.questionReview = this.createQuestionReviewStatsPage(
+        response.question_review_stats);
+    }
+  }
+
+  createTranslationContributionStatsPage(
+      stat: TranslationContributionBackendDict): StatsPage {
+    return {
+      data: [this.createTranslationContributionStat(stat)],
+      language: this.languageUtilService.getAudioLanguageDescription(
+        stat.language_code),
+      currentPageStartIndex: 0,
+      currentPageEndIndex: 5
+    };
+  }
+
+  createTranslationReviewStatsPage(
+      stat: TranslationReviewBackendDict): StatsPage {
+    return {
+      data: [this.createTranslationReviewStat(stat)],
+      language: this.languageUtilService.getAudioLanguageDescription(
+        stat.language_code),
+      currentPageStartIndex: 0,
+      currentPageEndIndex: 5
+    };
+  }
+
+  createQuestionContributionStatsPage(
+      stats: QuestionContributionBackendDict[]): StatsPage {
+    return {
+      data: stats.map((stat) => {
+        return this.createQuestionContributionStat(stat);
+      }),
+      currentPageStartIndex: 0,
+      currentPageEndIndex: 5
+    };
+  }
+
+  createQuestionReviewStatsPage(
+      stat: QuestionReviewBackendDict[]): StatsPage {
+    return {
+      data: stat.map((stat) => {
+        return this.createQuestionReviewStat(stat);
+      }),
+      currentPageStartIndex: 0,
+      currentPageEndIndex: 5
+    };
+  }
```

Comment:
I don't think you need a separate create method per stat type if you instead have a `createStatsPage(data, language?)` method.